x-pack/plugins/infra/public/components/asset_details/tabs/profiling/profiling_links.tsx

```diff
@@ -8,34 +8,32 @@
 import React from 'react';
 import { EuiFlexGroup, EuiFlexItem, EuiLink } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import { LocatorPublic } from '@kbn/share-plugin/common';
+import { SerializableRecord } from '@kbn/utility-types';
 import { HOST_FIELD } from '../../../../../common/constants';
-import { useKibanaContextForPlugin } from '../../../../hooks/use_kibana';
 
 const PROFILING_FEEDBACK_URL = 'https://ela.st/profiling-feedback';
 
-export type ProfilingPath = 'flamegraphs' | 'functions';
-
 interface Props {
   hostname: string;
-  profilingPath: ProfilingPath;
+  from: string;
+  to: string;
+  profilingLinkLocator: LocatorPublic<SerializableRecord>;
```

Comment:
What do you think of?
```suggestion
  profilingLinkLocator: FlamegraphLocator | TopNFunctionsLocator
```